### PR TITLE
Removed lists as default params, better filter error message

### PR DESCRIPTION
**Pull request by pandu-k**
filter string parse exception

* **Please check if the PR fulfills these requirements**
- [X] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes/features)
- [ ] Docs have been added / updated (for bug fixes / features)

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Refactor

* **What is the current behavior?** (You can also link to an open issue here)
Mutable objects in default params. This might cause [unexpected behaviour](https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments)

* **What is the new behavior (if this is a feature change)?**
Removed mutable objects in default params

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
No

* **Other information**:
Ran tox unit tests

### CODE REVIEW

File: src/marqo/tensor_search/tensor_search.py [hunk 1]
```diff
@@ -999,8 +1008,11 @@ def _vector_text_search(
         # KeyError indicates we have received a non-successful result
         try:
             if "index.max_result_window" in response["responses"][0]["error"]["root_cause"][0]["reason"]:
-                raise errors.IllegalRequestedDocCount("Marqo-OS rejected the response due to too many requested results. "
-                                             "Try reducing the query's limit parameter") from e
+                raise errors.IllegalRequestedDocCount(
+                    "Marqo-OS rejected the response due to too many requested results. "
+                    "Try reducing the query's limit parameter") from e
+            elif 'parse_exception' in response["responses"][0]["error"]["root_cause"][0]["reason"]:
```
Comment (Jeadie) [inline src/marqo/tensor_search/tensor_search.py:1014]: Is this related?

Comment (pandu-k) [reply src/marqo/tensor_search/tensor_search.py:1014]: Better error message: on multinode Marqo-os, the error message for bad filter string is slightly different. This helps handle it.

**Review by Jeadie — APPROVED** [review]
